src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.ts

```diff
@@ -177,6 +179,8 @@ export class NavigationComponent implements OnInit, OnDestroy {
       },
       () => {},
       () => {
+        // select default daemon from the selected cluster
+        this.rgwDaemonService.list().subscribe();
```

Comment:
Calling `subscribe()` but nothing is being done with the subscription, how does this work?

Reply:
@Pegonzal , if you see the `list()` method in rgw-daemon.service.ts, it selects a default daemon for the rgw api requests and this daemon's daemon name is added to the params of the api request in `request` method. We need to call this method here because what happens currently is suppose you are on the Objects -> Users page currently, when you switch over to some other cluster, it takes the selected daemon of the local cluster and sends it as the param for the api request of the second cluster and it results in 404 error in the second cluster as it doesn't recognize the daemon name of the local cluster. So we are calling this method on the cluster context switcher so on switching the cluster we select the daemon from the second cluster itself and prevent this error

Reply:
Thank you @aaSharma14 for the explanation.
Then, from my understanding you don't need to subscribe to the observable returned by `list()` since we are not doing anything with it.

Also, I find concerning that to select the default daemon we are calling ` this.rgwDaemonService.list()` as it does not have any indication that this changes the default daemon. Which of course leads to the need for the comment `select default daemon from the selected cluster`.

A lot of times comments are code smells for bad practices, and I believe this is one of them.
Ideally we would have something a long the lines of: `this.rgwDaemonService.updateDefault()`, which would remove the need for the comment and make the code cleaner.

Reply:
@Pegonzal ,
`Then, from my understanding you don't need to subscribe to the observable returned by list() since we are not doing anything with it.` - We actually still need to call the list method with subscribe since the subscription is required to trigger the request.

`A lot of times comments are code smells for bad practices, and I believe this is one of them.
Ideally we would have something a long the lines of: this.rgwDaemonService.updateDefault(), which would remove the need for the comment and make the code cleaner.` - I agree here but since this is existing implementation and its not in the scope of this PR to improve this, so I went on with adding a comment here at least to help debug why are we calling it here

Reply:
btw, removed this part from this PR, will open a separate PR for this change and probably move the multi-cluster part out of the navigation component. Thanks